lib/Models/Terria.ts

```diff
@@ -725,7 +822,16 @@ export default class Terria {
   updateParameters(parameters: ConfigParameters): void {
     Object.keys(parameters).forEach((key: string) => {
       if (this.configParameters.hasOwnProperty(key)) {
-        this.configParameters[key] = parameters[key];
+        if (key === "searchBar") {
+          // merge default and new
+          //@ts-ignore
+          this.configParameters[key] = {
+            ...this.configParameters[key],
+            ...parameters[key]
+          };
+        } else {
```

Comment:
merging searchBar params so specified defaults don't get overridden with undefined. We might even run deep merge on config parameters something as `_.merge(this.configParameters, parameters)`